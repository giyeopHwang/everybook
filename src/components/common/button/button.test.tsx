import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Button from './button.component';

describe('Button', () => {
  const mock = vi.fn();

  beforeEach(() => {
    render(<Button onClick={mock}>foo</Button>);
  });

  afterEach(() => {
    vi.restoreAllMocks();
  });

  it('renders a Button component', async () => {
    const button = screen.getByText('foo');
    expect(button).toMatchSnapshot();
  });

  it('fires callback when clicked', async () => {
    const button = screen.getByText('foo');
    fireEvent.click(button);
    expect(mock).toHaveBeenCalledOnce();
  });
});
