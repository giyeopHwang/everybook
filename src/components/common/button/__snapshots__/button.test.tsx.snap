// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > renders a Button component 1`] = `
<button
  class="sc-bcXGCL hyeVjf"
>
  foo
</button>
`;
